toolkit/tools/pkg/imagecustomizerlib/customizepackages_rpm.go

```diff
@@ -0,0 +1,98 @@
+package imagecustomizerlib
+
+import (
+	"context"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+)
+
+// managePackagesRpm provides a shared implementation for RPM-based package management
+func managePackagesRpm(ctx context.Context, buildDir string, baseConfigPath string, config *imagecustomizerapi.OS,
+	imageChroot *safechroot.Chroot, toolsChroot *safechroot.Chroot,
+	rpmsSources []string, useBaseImageRpmRepos bool, snapshotTime string, distroConfig distroHandler,
+) error {
+	var err error
+
+	packageManagerChroot := imageChroot
+	if toolsChroot != nil {
+		packageManagerChroot = toolsChroot
+	}
+
+	// Get package manager from distribution config
+	pmHandler := distroConfig.getPackageManager()
+
+	// Setup distribution-specific configuration if needed
+	if distroConfig.getDistroName() == distroNameAzureLinux {
+		// Setup Azure Linux specific TDNF configuration with snapshot
+		err = createTempTdnfConfigWithSnapshot(packageManagerChroot, imagecustomizerapi.PackageSnapshotTime(snapshotTime))
```

Comment:
An error should result if `snapshotTime` is provided but the distro doesn't support it.